### PR TITLE
Mipmap generation for GL

### PR DESCRIPTION
**Pull request by snuk182**

### CODE REVIEW

File: src/backend/gl/src/tex.rs [hunk 1]
```diff
@@ -404,45 +411,31 @@ pub fn make_without_storage(gl: &gl::Gl, desc: &t::Info, cty: ChannelType) ->
         Ok(t) => t,
         Err(_) => return Err(t::CreationError::Format(desc.format, Some(cty))),
     };
-
+    
+    let levels = get_num_levels(desc, mip);
     let fixed_loc = desc.bind.contains(SHADER_RESOURCE);
-    make_widout_storage_impl(gl, desc.kind, gl_format, gl_pixel_format, gl_data_type,
-                             desc.levels, fixed_loc)
+    make_without_storage_impl(gl, desc.kind, gl_format, gl_pixel_format, gl_data_type,
```
Comment (kvark) [inline src/backend/gl/src/tex.rs]: I missed this last time, but this is not how we are supposed to determine the level count.
It *always* comes from `t::Info`, unconditionally.
What your new parameter is supposed to change is how the data slices are being treated.

File: src/core/src/factory.rs [hunk 2]
```diff
@@ -330,7 +331,7 @@ pub trait Factory<R: Resources> {
         -> Result<handle::RawDepthStencilView<R>, TargetViewError>;
 
     fn create_texture<S>(&mut self, kind: texture::Kind, levels: target::Level,
-                      bind: Bind, usage: Usage, channel_hint: Option<format::ChannelType>)
+                      bind: Bind, usage: Usage, channel_hint: Option<format::ChannelType>)
```
Comment (kvark) [inline src/core/src/factory.rs]: extra spaces at the end?

File: src/backend/gl/src/tex.rs (hunk 1 above)
Comment (snuk182) [reply src/backend/gl/src/tex.rs]: Did not get it. With `Mipmap::Generated` we *change* the levels count, because it is actually changed.

Comment (kvark) [reply src/backend/gl/src/tex.rs]: You need to understand what a texture is. It's a number of 2D texel buffers, basically, allocated in VRAM. That allocation is precisely described by the `Info` struct. You do **not** need to change it.
What you are doing is saying: hey, I'm providing that texel data, but it's only for the base mipmap level, and I want the other levels to be left uninitialized. That's why I asked you to move the parameter into the data `Option`.

Comment (snuk182) [reply src/backend/gl/src/tex.rs]: Ok, so what should be instead then?

**Comment by snuk182** [comment]
Next iteration comes shortly.

**Pull request closed without merging**